### PR TITLE
JBIDE-21865 Fix jbosstools-server build from maste

### PR DESCRIPTION
**Pull request by dgolovin**
Migrate to new API

### CODE REVIEW

File: wtp/plugins/org.jboss.tools.wtp.server.launchbar/src/org/jboss/tools/wtp/server/launchbar/ModuleLaunchConfigurationProvider.java [hunk 1]
```diff
@@ -54,13 +55,13 @@ public boolean supports(ILaunchDescriptor descriptor, IRemoteConnection target)
 
 	
 	@Override
-	public ILaunchConfigurationType getLaunchConfigurationType(ILaunchDescriptor descriptor, IRemoteConnection target)
+	public ILaunchConfigurationType getLaunchConfigurationType(ILaunchDescriptor descriptor, ILaunchTarget arg1)
```
Comment (fbricon) [inline wtp/plugins/org.jboss.tools.wtp.server.launchbar/src/org/jboss/tools/wtp/server/launchbar/ModuleLaunchConfigurationProvider.java]: , ILaunchTarget target)

File: wtp/plugins/org.jboss.tools.wtp.server.launchbar/src/org/jboss/tools/wtp/server/launchbar/ModuleLaunchConfigurationProvider.java [hunk 2]
```diff
@@ -129,4 +125,10 @@ public boolean launchConfigurationChanged(ILaunchConfiguration configuration) th
 		return false;
 	}
 
+	@Override
+	public void launchTargetRemoved(ILaunchTarget arg0) throws CoreException {
```
Comment (fbricon) [inline wtp/plugins/org.jboss.tools.wtp.server.launchbar/src/org/jboss/tools/wtp/server/launchbar/ModuleLaunchConfigurationProvider.java]: , ILaunchTarget target)

File: wtp/plugins/org.jboss.tools.wtp.server.launchbar/src/org/jboss/tools/wtp/server/launchbar/ModuleLaunchConfigurationProvider.java [hunk 3]
```diff
@@ -129,4 +125,10 @@ public boolean launchConfigurationChanged(ILaunchConfiguration configuration) th
...
+		// TODO Auto-generated method stub
```
Comment (fbricon) [inline wtp/plugins/org.jboss.tools.wtp.server.launchbar/src/org/jboss/tools/wtp/server/launchbar/ModuleLaunchConfigurationProvider.java]: remove comment

**Comment by dgolovin** [comment]
updated

**Comment by robstryker** [comment]
I'll review this today. Patch looks ok on glance but I need to test it

**Comment by robstryker** [comment]
Submitted https://github.com/dgolovin/jbosstools-server/pull/2   as a more complete fix

**Comment by dgolovin** [comment]
Local build is happy with this PR.
+1

**Comment by robstryker** [comment]
squashed and rebased and pushed to master

**Pull request closed without merging**